pds_doi_core/input/pds4_util.py

```diff
@@ -60,25 +95,41 @@ def process_pds4_fields(self, pds4_fields):
                                                 requests.utils.quote(pds4_fields['vid']))
         editors = self.get_editor_names(pds4_fields['editors'].split(';')) if 'editors' in pds4_fields.keys() else None
 
+
+        # The 'authors' field is inconsistent.  Try to make a best guess on which method is better.
+        o_best_method =  self._make_best_guess_method_to_parse_authors(pds4_fields['authors'])
+        if o_best_method == BestParserMethod.BY_COMMA:
+            authors_list = pds4_fields['authors'].split(',')
+        elif o_best_method == BestParserMethod.BY_SEMI_COLON:
+            authors_list = pds4_fields['authors'].split(';')
+        else:
+            logger.error(f"o_best_method,pds4_fields['authors'] {o_best_method,pds4_fields['authors']}")
+            raise InputFormatException("Cannot split the authors using comma or semi-colon.")
+
         doi = Doi(title=pds4_fields['title'],
                   description=pds4_fields['description'],
                   publication_date=self.get_publication_date(pds4_fields),
                   product_type=product_type,
                   product_type_specific= "PDS4 Refereed Data " + product_type,
                   related_identifier=pds4_fields['lid'] + '::' + pds4_fields['vid'],
                   site_url=site_url,
-                  authors=self.get_author_names(pds4_fields['authors'].split(',')),
+                  authors=self.get_author_names(authors_list),
                   editors=editors,
                   keywords=self.get_keywords(pds4_fields)
                   )
 
+        # Add field 'date_record_added' because the XSD requires it.
+        if doi.date_record_added is None:
```

Comment:
Looking in the pds_util.py function, there isn't any where 'date_record_added" is parsed for.

To not be misleading, the 'if' statement has been removed.

        # Add field 'date_record_added' because the XSD requires it.
        doi.date_record_added = datetime.now().strftime('%Y-%m-%d')

Reply:
I realise now that we should get the creation date from the transaction database in case we update a record. I will create a ticket for that.